content/blog/2017/2017-02-07-declarative-maven-project.adoc

```diff
@@ -0,0 +1,295 @@
+I'm going to setup a Multi-branch Pipeline and point it at my repository.
+
+image::/images/post-images/2017-02-07/junit-config.png[JUnit Multi-branch Pipeline Configuration, role="center"]
+
+I've also set this Pipeline's Git configuration to automatically "clean after
+checkout" and to only keep the ten most recent runs.
+
+== Writing a Minimal Pipeline
+
+As has been said before, Declarative Pipeline provides a more structured,
+"opinionated" way to create Pipelines. I'm going to start by creating a minimal
+Declarative Pipeline and adding it to my branch.  Below is a minimal Pipeline
+(with annotations) that just prints a message:
+
+[pipeline]
+----
+// Declarative //
+pipeline { // <1>
+    agent any // <2> <3>
+    stages { // <4>
+        stage('Build') { // <5>
+            steps { // <6>
+               echo 'This is a minimal pipeline.' // <7>
+            }
+        }
+    }
+}
+// Scripted //
+node { // <2>
+    checkout scm // <3>
+    stage ('Build') { // <4>
+        echo 'This is a minimal pipeline.' // <6>
+    }
+}
+----
+<1> All Declarative Pipelines start with a `pipeline` section.
+<2> Select where to run this Pipeline, in this case "any" agent, regardless of label.
+<3> Declarative automatically performs a checkout of source code on the agent,
+    whereas Scripted Pipeline users must explicitly call `checkout scm`,
+<4> A minimal Declarative Pipeline is defined a series of stages.
+<5> Run the "Build" stage.
+<6> Each stage in a Declarative Pipeline runs a series of steps.
+<7> Run the `echo` step to print a message in the Console Output.
+
+NOTE: If you are familiar with Scripted Pipeline, you can toggle the above
+Declarative code sample to show the Scripted equivalent.
+
+Once I add the Pipeline above to my `Jenkinsfile` and run "Branch Indexing", my
+Jenkins will pick it up and run run it.  We see that the Declarative Pipeline
+has added stage called "Declarative: Checkout SCM":
+
+image::/images/post-images/2017-02-07/junit-minimal.png[Minimal Declarative Pipeline, role="center"]
+
+This a "dynamic stage", one of several the kinds that Declarative Pipeline
+adds as needed for clearer reporting.
+In this case, it is a stage in which the Declarative Pipeline automatically does`checkout scm` on the main agent.
+As you can see above, we didn't have to tell it do any of this,
+
+.Console Output
+[source]
+----
+[Pipeline] node
+Running on osx_mbp in /Users/bitwiseman/jenkins/agents/osx_mbp/workspace/blog_add-declarative-pipeline
+[Pipeline] {
+[Pipeline] stage
+[Pipeline] { (Declarative: Checkout SCM)
+[Pipeline] checkout
+Cloning the remote Git repository
+{ ... truncated 20 lines ... }
+[Pipeline] }
+[Pipeline] // stage
+[Pipeline] stage
+[Pipeline] { (Build)
+[Pipeline] echo
+This is a minimal pipeline
+[Pipeline] }
+[Pipeline] // stage
+[Pipeline] }
+[Pipeline] // node
+[Pipeline] End of Pipeline
+Finished: SUCCESS
+----
+
+Declarative Pipeline code is a little more verbose than the equivalent Scripted Pipeline,
```

Comment:
Very minor: Maybe refer to it as "syntax" rather than "code"?
(though we do call it "pipeline as code"…)